src/DssSpell.sol

```diff
@@ -65,312 +70,293 @@ contract DssSpellAction is DssAction {
+        _transferUsds(CLOAKY_ENNOIA,   20_110 * WAD);
+
+        // BLUE - 50,167 USDS - 0xb6C09680D822F162449cdFB8248a7D3FC26Ec9Bf
+        _transferUsds(BLUE,            50_167 * WAD);
+
+        // BLUE - 330,000 SKY - 0xb6C09680D822F162449cdFB8248a7D3FC26Ec9Bf
+        _transferSky( BLUE,           330_000 * WAD);
 
         // ---------- Spark Proxy Spell ----------
-        // Forum: https://forum.sky.money/t/14-nov-2024-proposed-changes-to-spark-for-upcoming-spell/25466
-        // Poll: https://vote.makerdao.com/polling/QmQizL1F
-        // Poll: https://vote.makerdao.com/polling/Qmbohkr5
-        // Poll: https://vote.makerdao.com/polling/QmYqM8Yf
-        // Poll: https://vote.makerdao.com/polling/QmXsXzot
-        // Poll: https://vote.makerdao.com/polling/Qmf955yA
-
-        // Execute Spark Proxy Spell at 0x8a3aaeAC45Cf3D76Cf82b0e4C63cCfa8c72BDCa7
+        // Forum: https://forum.sky.money/t/stability-scope-parameter-changes-18-sfs-dsr-ssr-spark-effective-dai-borrow-rate-spark-liquidity-layer/25593
+        // Forum: https://forum.sky.money/t/28-nov-2024-proposed-changes-to-spark-for-upcoming-spell/25543
+        // Poll: https://vote.makerdao.com/polling/QmSxJJ6Z
+        // Poll: https://vote.makerdao.com/polling/QmaxFZfF
+        // Poll: https://vote.makerdao.com/polling/QmWUkstV
+        // Poll: https://vote.makerdao.com/polling/QmQ2Umfm
+        // Poll: https://vote.makerdao.com/polling/QmcNd4mH
+
+        // Note: The spark spell below expects the new auto-line settings to be effective.
+        DssAutoLineAbstract(AUTO_LINE).exec("ALLOCATOR-SPARK-A");
+
+        // Execute Spark Proxy Spell at 0x6c87D984689CeD0bB367A58722aC74013F82267d
         ProxyLike(SPARK_PROXY).exec(SPARK_SPELL, abi.encodeWithSignature("execute()"));
     }
+
+    /// @notice wraps the operations required to transfer USDS from the surplus buffer.
+    /// @param usr The USDS receiver.
+    /// @param wad The USDS amount in wad precision (10 ** 18).
+    function _transferUsds(address usr, uint256 wad) internal {
+        // Note: Enforce whole units to avoid rounding errors
+        require(wad % WAD == 0, "transferUsds/non-integer-wad");
+        // Note: DssExecLib currently only supports Dai transfers from the surplus buffer.
+        DssExecLib.sendPaymentFromSurplusBuffer(address(this), wad / WAD);
+        // Note: Approve DAI_USDS for the amount sent to be able to convert it.
+        DAI.approve(DAI_USDS, wad);
+        // Note: Convert Dai to USDS for `usr`.
+        DaiUsdsLike(DAI_USDS).daiToUsds(usr, wad);
+    }
+
+    /// @notice wraps the operations required to transfer SKY from the treasury.
+    /// @param usr The SKY receiver.
+    /// @param wad The SKY amount in wad precision (10 ** 18).
+    function _transferSky(address usr, uint256 wad) internal {
+        // Note: Enforce exact conversion to avoid rounding errors
+        require(wad % MKR_SKY_RATE == 0, "transferSky/non-exact-conversion");
```

Comment:
Similar to the above, we can have a function that only expects SKY in whole units as an input (as it is 240 times less likely than MKR to need decimals). 

If you do so, please add additional `require(_amount < WAD)` to prevent unit mistakes (like [sendPaymentFromSurplusBuffer does](https://github.com/makerdao/dss-exec-lib/blob/69b658f35d8618272cd139dfc18c5713caf6b96b/src/DssExecLib.sol#L1079))

Reply:
Answered above.